Cypress commands and runs improvements (#3242)
plone/volto: docs/source/developer-guidelines/acceptance-tests.md

  "keywords": "Volto, Plone, frontend, React, helper command, redux, acceptance, tests, Cypress"
---

# Acceptance testing

Volto uses [Cypress](https://www.cypress.io/) for browser-based acceptance testing.

There are a number of fixtures available covering all the configuration use cases. These fixtures have both a specific backend and frontend configuration setup and a related set of tests. The CI infrastructure runs them all automatically on every push to a branch or PR.

The tests can be run in headless mode (same as the CI does), or within the Cypress user interface. The latter is the one that you run under development.

## How to run acceptance tests locally (during development)

When writing new acceptance tests, you usually want to minimize the time it takes to run the tests, while also being able to debug or inspect what's going on.

> Being able to restart individual components also come in handy, so it's recommended to start three individual terminal sessions for running the Plone backend, the Volto frontend, and the acceptance tests.

Reviewer: ```suggestion
Being able to restart individual components also comes in handy.
It's recommended to start three individual terminal sessions, one each for running the Plone backend, the Volto frontend, and the acceptance tests.
```

> 1. Run the backend fixture

Reviewer: ```suggestion
1.  Run the backend fixture.
```

    ```shell
    make start-test-acceptance-server
    ```

> 2. Run the frontend fixture

Reviewer: ```suggestion
2.  Run the frontend fixture.
```

    ```shell
    make start-test-acceptance-frontend
    ```

> 3. Run the Cypress tests for that fixture

Reviewer: ```suggestion
3.  Run the Cypress tests for that fixture.
```

    ```shell
    make test-acceptance
    ```

Available fixtures:

> - Core (core or not special naming in the test commands)

Reviewer: ```suggestion
- Core (`core` or not special naming in the test commands)
```

> - Multilingual (multilingual)

Reviewer: ```suggestion
- Multilingual (`multilingual`)
```

> - Working Copy (workingCopy)

Reviewer: ```suggestion
- Working Copy (`workingCopy`)
```

> - Core Sandbox (coresandbox)

Reviewer: ```suggestion
- Core Sandbox (`coresandbox`)
```
